src/algebra/finite_map/mod.rs

```diff
@@ -178,8 +195,48 @@ where
         self.value.get(key).cloned().unwrap_or_else(Value::zero)
     }
 
-    fn support<'a>(&self) -> SupportIter<'_, Key, Value> {
-        self.value.keys()
+    fn get(&self, key: &Key) -> Option<&Value> {
+        self.value.get(key)
+    }
+
+    fn update<F>(&mut self, key: &Key, f: F)
+    where
+        F: FnOnce(&mut Value),
+    {
+        match self.value.raw_entry_mut().from_key(key) {
+            RawEntryMut::Occupied(mut oe) => {
+                let val = oe.get_mut();
+                f(val);
+                if val.is_zero() {
+                    oe.remove();
+                }
+            }
+            RawEntryMut::Vacant(ve) => {
+                let mut val = Value::zero();
+                f(&mut val);
+                ve.insert(key.clone(), val);
```

Comment:
we should not store val if it is zero

Reply:
Right!  I am also fixing a similar issue in `ZSetHashMap::singleton()`.